infrastructure/src/main/java/org/corfudb/infrastructure/SequencerServerCache.java

```diff
@@ -167,15 +126,25 @@ public long size() {
      * @param newTail        global tail
      */
     public void put(ConflictTxStream conflictStream, long newTail) {
-        conflictCache.put(conflictStream, newTail);
+        if (cacheHash.size() == cacheSize) {
+            ConflictTxStreamEntry entry = cacheEntries.peek();
+            invalidateUpTo(firstAddress() + 1);
+        }
+
+        ConflictTxStreamEntry entry = new ConflictTxStreamEntry(conflictStream, newTail);
+        cacheEntries.add(entry);
+        cacheHash.put(conflictStream, entry.txVersion);
+        maxAddress = Math.max(newTail, maxAddress);
     }
 
     /**
      * Discard all entries in the cache
      */
     public void invalidateAll() {
-        log.info("Invalidate sequencer server cache");
-        conflictCache.invalidateAll();
+        log.info("Invalidate all entries in sequencer server cache and update maxConflictWildcard to {}", maxAddress);
```

Comment:
Since you have replaced caffeine cache by a hashmap, would be great if you also get rid of this method.
Instead of spending linear time O(n) on cleaning maps, you can just create a new instance of a SequencerServerCache and let garbage collector do his work

Reply:
+1, I would also suggest to get rid of this method. The caller can recreate these objects.